support/cas-server-support-duo-core/src/main/java/org/apereo/cas/adaptors/duo/authn/DuoAuthenticationHandler.java

```diff
@@ -97,6 +98,10 @@ private AuthenticationHandlerExecutionResult authenticateDuoCredential(final Cre
 
     @Override
     public boolean supports(final Credential credential) {
-        return provider.validateMark(credential.getMark());
+        if (credential instanceof MultifactorAuthenticationCredential) {
+            final String id = ((MultifactorAuthenticationCredential) credential).getProviderId();
+            return provider.validateId(id);
+        }
+        return false;
```

Comment:
Isn't this more like a catastrophic error, as an exception?

Reply:
No, this gets called as part of PolicyBasedAuthManger handler resolve for finding a handler that can auth the credential. So it could get passed Credentials of any kind, also MFA credentials created by other MFA providers than the one used by this one.